google/cloud/storage/object_metadata_test.cc

```diff
@@ -0,0 +1,128 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/storage/object_metadata.h"
+#include <gmock/gmock.h>
+
+/// @test Verify that we parse JSON objects into ObjectMetadata objects.
+TEST(ObjectMetadataTest, Parse) {
+  std::string text = R"""({
+      "kind": "storage#object",
```

Comment:
For readability, would you mind alphabetizing these keys?

Reply:
Done.